airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -359,7 +361,9 @@ def parse_response(
         records = self.record_selector.select_records(
             response=response, stream_state=self.state, stream_slice=stream_slice, next_page_token=next_page_token
         )
-        self._last_records = records
+        self._records_from_last_response = records
+        if records:
+            self._latest_record = records[-1]
```

Comment:
Apparently, the typing for `record_selector.select_records` is not good since https://github.com/airbytehq/airbyte/pull/24772 and therefore this code does not work

Reply:
This was reverted as part of https://github.com/airbytehq/airbyte/pull/27789 and we can not fetch using an index